tests/cpp/target_test.cc

```diff
@@ -289,6 +291,45 @@ TEST(TargetCreation, ProcessStrings) {
   ASSERT_EQ(array7[1][1][0], "fred");
 }
 
+TEST(TargetCreation, LLVMCommandLineFatal) {
```

Comment:
just documenting offline convo: i think it would improve the PR to add a few more unit tests. @kparzysz-quic pointed out this test case contains several examples of bad options; action item is to move the various "bad options" in this test case into separate test cases to increase coverage